test/libsolidity/util/compiler/CompilerHost.h

```diff
@@ -0,0 +1,48 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <test/libsolidity/util/compiler/Compiler.h>
+#include <test/libsolidity/util/compiler/InternalCompiler.h>
+
+namespace solidity::frontend::test
+{
+
+class CompilerHost
+{
+public:
+    /// Takes the current compiler input, requests the hosted compiler to compile
+    /// and updates the output accordingly.
+    CompilerOutput const& compile(CompilerInput _input);
```

Comment:
Do you expect `compile` to be called multiple times before `output` is requested? Otherwise you could get rid of `reset` as well as `output`, don't even store it here, and just leave it at `compile`.

Reply:
No, but the gas cost tests need to request the output independently from the actual compilation. Ultimately, the gas costs test should be moved out, which would loosen this requirement. But for time being, @ekpyron and me decided to leave the tests as is and support the independent output request.

Reply:
The output could be saved in the execution framework right? But fair enough, it's better not to change too many things at once.